mnet/mnet_test.go

```diff
@@ -52,6 +54,42 @@ func TestNetwork(t *testing.T) {
 		}
 	})
 
+	t.Run("ListenRandom", func(t *testing.T) {
+		synctest.Test(t, func(t *testing.T) {
+			n := mnet.New(t.Name())
+			lst, err := n.Listen("tcp", "foo:0")
+			if err != nil {
+				t.Fatalf("Listen: unexpected error: %v", err)
+			}
+			checkHostPort(t, lst.Addr().String(), "foo")
+
+			// Verify that we can actually dial the assigned address.
+			go func() {
+				cli, err := lst.Accept()
+				if err != nil {
+					t.Errorf("Accept: unexpected error: %v", err)
+				}
+				t.Logf("Connection to %q accepted from %q OK", cli.LocalAddr(), cli.RemoteAddr())
+				cli.Close()
+			}()
+
+			srv, err := n.Dial("tcp", lst.Addr().String())
+			if err != nil {
+				t.Fatalf("Dial %q: unexpected error: %v", lst.Addr(), err)
+			}
+			t.Logf("Connected to %q OK", srv.RemoteAddr())
+			srv.Close()
+
+			synctest.Wait() // allow logging to finish
```

Comment:
FWIW I _think_ this is a no-op. It allows us to wait until all goroutines are durably blocked, which is useful if we want to do something afterwards, but in this case we don't want to do anything afterwards, and synctest.Test will panic if there _are_ any goroutines durably blocked...